### ISSUE TITLE
Lazy cyclic dependencies are no longer possible in v2.0

### ISSUE THREAD

**Issue by Yves-Be** [DevTeam/Pure.DI#32]
### Description
Since v2.0, it is no longer possible to have a cyclic dependency where one of the dependencies is lazy resolved.
this results in the following error:  
> error DIE001: A cyclic dependency has been found Pure.DI.Example.Dependency <-- System.Lazy<Pure.DI.Example.IDependency2> <-- System.Func<Pure.DI.Example.IDependency2> <-- Pure.DI.Example.Dependency2 <-- Pure.DI.Example.Dependency.

In v1, code was generated to resolve the lazy dependency lazily.

### Example
```
public class Dependency : IDependency
{
    public Dependency(Lazy<IDependency2> dependency2) { }
}

public class Dependency2 : IDependency2
{
    public Dependency2(IDependency dependency) { }
}

public partial class Composition
{
    public static void Setup() =>
        DI.Setup(nameof(Composition))
            .Bind<IDependency>().To<Dependency>()
            .Bind<IDependency2>().To<Dependency2>()
            .Bind<IService>().To<Service>().Root<IService>("Root");
}
```

### Repository
https://github.com/Yves-Be/Pure.DI.LazyCircularDependency

**Comment by NikolayPianikov** [comment]
Unfortunately, it's a compromise to make the generation faster. I will try to find a solution. Right now there is an easy way to solve this problem:

```c#
Bind<Func<TT>>().To(_ => new Func<TT>(Resolve<TT>))
```

so your example might look like this:

```c#
using Pure.DI;

var service = new Composition().Root;

return;

public class Dependency : IDependency
{
    private readonly Lazy<IDependency2> _dependency2;
    public Dependency(Lazy<IDependency2> dependency2) => _dependency2 = dependency2;
}

public interface IDependency    
{
}

public class Dependency2 : IDependency2
{
    private readonly IDependency _dependency;
    public Dependency2(IDependency dependency) => _dependency = dependency;
}

public interface IDependency2   
{
}

public interface IService
{
}

public class Service : IService
{
    private readonly IDependency2 _dependency2;

    public Service(IDependency2 dependency2) => _dependency2 = dependency2;
}

public partial class Composition
{
    public void Setup() =>
        DI.Setup(nameof(Composition))
            .Bind<Func<TT>>().To(_ => new Func<TT>(Resolve<TT>))
            .Bind<IDependency>().To<Dependency>()
            .Bind<IDependency2>().To<Dependency2>()
            .Bind<IService>().To<Service>().Root<IService>("Root");
}
```

**Comment by NikolayPianikov** [comment]
Unfortunately, in this case performance degrades and _PerResolve_ lifetime stops working.
So this behavior will not be fixed at the moment. It is recommended to avoid cyclic dependencies.

**Comment by NikolayPianikov** [comment]
@Yves-Be could you please try [this version](https://www.nuget.org/packages/Pure.DI/2.0.13-beta)

**Comment by Yves-Be** [comment]
That seems to work, up until you inject IDependency in the Service.
I believe it is good enough for my case though.

**Comment by NikolayPianikov** [comment]
@Yves-Be yeah, I got it, I'll fix it so it's fully functional.

**Comment by NikolayPianikov** [comment]
Fixed in [2.0.13](https://www.nuget.org/packages/Pure.DI/2.0.13)
